tests/csapi/room_typing_test.go

```diff
@@ -40,3 +40,55 @@ func TestTyping(t *testing.T) {
 		return false
 	}))
 }
+
+// sytest: Typing notifications don't leak
+func TestLeakyTyping(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintOneToOneRoom)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+	bob := deployment.Client(t, "hs1", "@bob:hs1")
+	charlie := deployment.RegisterUser(t, "hs1", "charlie", "charliepassword", false)
+
+	roomID := alice.CreateRoom(t, map[string]interface{}{"preset": "public_chat"})
+	bob.JoinRoom(t, roomID, nil)
+
+	bobToken := bob.MustSyncUntil(t, client.SyncReq{}, client.SyncJoinedTo(bob.UserID, roomID))
+
+	_, charlieToken := charlie.MustSync(t, client.SyncReq{TimeoutMillis: "0"})
+
+	alice.MustDoFunc(t, "PUT", []string{"_matrix", "client", "v3", "rooms", roomID, "typing", alice.UserID}, client.WithJSONBody(t, map[string]interface{}{
+		"typing":  true,
+		"timeout": 10000,
+	}))
+
+	bob.MustSyncUntil(t, client.SyncReq{Since: bobToken}, client.SyncEphemeralHas(roomID, func(result gjson.Result) bool {
+		if result.Get("type").Str != "m.typing" {
+			return false
+		}
+		for _, item := range result.Get("content").Get("user_ids").Array() {
+			if item.Str == alice.UserID {
+				return true
+			}
+		}
+		return false
+	}))
+
+	res, _ := charlie.MustSync(t, client.SyncReq{TimeoutMillis: "2000", Since: charlieToken})
```

Comment:
```suggestion
	// Charlie is not in the room, so should not see Alice typing.
	res, _ := charlie.MustSync(t, client.SyncReq{TimeoutMillis: "2000", Since: charlieToken})
```